pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -1044,72 +1045,88 @@ protected void internalDeleteTopic(boolean authoritative, boolean deleteSchema)
+                                        for (int i = 0; i < partitionMetadata.partitions; i++) {
+                                            existsFutures.put(i,
+                                                    topicResources().persistentTopicExists(topicName.getPartition(i)));
+                                        }
+                                        FutureUtil.waitForAll(Lists.newArrayList(existsFutures.values()))
+                                                .thenApply(__ ->
+                                                existsFutures.entrySet().stream().filter(e -> e.getValue().join())
+                                                        .map(item -> topicName.getPartition(item.getKey()).toString())
+                                                        .collect(Collectors.toList())
+                                        ).thenAccept(topics -> {
+                                            if (log.isDebugEnabled()) {
+                                                log.debug("activeTopics : {}", topics);
+                                            }
+                                            topics.forEach(topic -> {
+                                                try {
+                                                    CompletableFuture<List<String>> subscriptionsAsync = pulsar()
+                                                            .getAdminClient()
+                                                            .topics().getSubscriptionsAsync(topic);
+                                                    subscriptionFutures.add(subscriptionsAsync
+                                                            .thenApply(subscriptions::addAll));
+                                                } catch (PulsarServerException e) {
+                                                    throw new RestException(e);
+                                                }
+                                            });
+                                        }).thenAccept(__ -> resumeAsyncResponse(asyncResponse,
+                                                        subscriptions, subscriptionFutures));
+                                    } else {
+                                        for (int i = 0; i < partitionMetadata.partitions; i++) {
+                                            CompletableFuture<List<String>> subscriptionsAsync = pulsar()
+                                                    .getAdminClient().topics()
+                                                    .getSubscriptionsAsync(topicName.getPartition(i).toString());
+                                            subscriptionFutures.add(subscriptionsAsync
+                                                    .thenApply(subscriptions::addAll));
+                                        }
+                                        resumeAsyncResponse(asyncResponse, subscriptions, subscriptionFutures);
```

Comment:
Interesting, we have a `resumeAsyncResponse` method but looks like can only be used by list subscriptions.

Reply:
emmm, do we need to move this method back ?

Reply:
We can consider if the `resumeAsyncResponse` can be changed into a general method, not only for list subscriptions.  If we can't have a general method to handle the response, it's better to hand it over to a specific method to handle the response by themselves. 

It's not the scope of this PR.

Reply:
Well, I think I need to clean up the code after the #13854 work is done (like checking for duplicates, refactoring some code, etc.).